gordo_components/model/models.py

```diff
@@ -194,20 +155,30 @@ def get_params(self, **params):
 
     def __call__(self):
         build_fn = register_model_builder.factories[self.__class__.__name__][self.kind]
-        with possible_tf_mgmt(self):
-            return build_fn(**self.sk_params)
+        return build_fn(**self.sk_params)
 
     def save_to_dir(self, directory: str):
         params = self.get_params()
         with open(path.join(directory, "params.json"), "w") as f:
             json.dump(params, f)
         if hasattr(self, "model") and self.model is not None:
-            with possible_tf_mgmt(self):
-                self.model.save(path.join(directory, "model.h5"))
-                if hasattr(self.model, "history") and self.model.history is not None:
-                    f_name = path.join(directory, "history.pkl")
-                    with open(f_name, "wb") as history_file:
-                        pickle.dump(self.model.history, history_file)
+            with open(path.join(directory, "model.json"), "w") as model_json:
+                json.dump(self.model.to_json(), model_json)
```

Comment:
this is weird, why do you have to do json.dump on a json?

Reply:
Also, do you want to do a quick benchmark of how long it takes to load a (big lstm) tensorflow model with the 5 alternatives:
- `model.save('path_to_saved_model', save_format='tf', include_optimizer=True)`
- `model.save('path_to_saved_model', save_format='h5', include_optimizer=True)`
- `model.save('path_to_saved_model', save_format='tf', include_optimizer=False)`
- `model.save('path_to_saved_model', save_format='h5', include_optimizer=False)`
- save model weight/arcitecture as json.

My guess is that the 3 last ones are pretty much the same.

Reply:
Results are in!
```
Save as JSON and h5 seperatly
-----
14.13s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
12.60s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
1.50s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.tf, include_optimizer=True, compile=True
-----
51.24s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
50.07s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
3.73s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.h5, include_optimizer=True, compile=True
-----
16.96s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
14.52s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
2.02s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.h5, include_optimizer=True, compile=False
-----
12.30s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
12.12s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
1.56s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.h5, include_optimizer=False, compile=False
-----
12.62s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
12.35s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
2.03s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]
```

Note: It was not possible to have `save_format='tf'` with `compile=False` _and/or_ `include_optimizer=False`